### DISCUSSION TITLE
dynamically turn off and on  validation

### DISCUSSION THREAD

**Discussion by aman-275** [react-hook-form/react-hook-form#5866] [Q&A]
in my form i am hiding field on and off using usestate so when i hide field after it has been use and resetting it values field validation doesnt let it get submit empty ..is there a way  i can dynamically turn on and off validation

**Comment by bluebill1049** [comment]
use `validate` function, you can use external state inside or use `getValues` to retrieve other input value to do conditional logic.

**Reply by aman-275** [reply]
I am use react  native..my usecase is when user select true then the field will be displayed  and when user false then the field will not display.. i have 2 buttons add and submit.... field means any text field so user seleft true he writes  something  on field and click add button then next time he select  false but now he clicks on add button and i have set rules reqired true so validation  failed now i cant add...i dont wanna  use custom validation library like yup...any usecase around it like unregistering the field when user select false
